Update Vector's uninitializedCopy() to use std::span

cdumez opened the pull request:
#### af23282667330ccad8d2566c0aa09bcbad49a3d0
<pre>
Update Vector&apos;s uninitializedCopy() to use std::span
<a href="https://bugs.webkit.org/show_bug.cgi?id=304756">https://bugs.webkit.org/show_bug.cgi?id=304756</a>

Reviewed by Keith Miller.

* Source/JavaScriptCore/yarr/generateYarrUnicodePropertyTables.py:
(PropertyData.convertStringToCppFormat):
* Source/WTF/wtf/Vector.h:
(WTF::VectorCopier::uninitializedCopy):
(WTF::VectorTypeOperations::uninitializedCopy):
(WTF::Vector::Vector):
(WTF::Malloc&gt;::Vector):
(WTF::=):
(WTF::Malloc&gt;::append):
(WTF::Malloc&gt;::unsafeAppendWithoutCapacityCheck):
(WTF::Malloc&gt;::insertSpan):

Canonical link: <a href="https://commits.webkit.org/305031@main">https://commits.webkit.org/305031@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/259751a3cf27aca1ef7860a07b82b6f812cf95a7

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/137229 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/9589 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/48516 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/144980 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/90202 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/0d227df8-470c-47c9-a233-657256ca01a5) 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/10292 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/9716 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/104949 "Passed tests") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/60/builds/76535 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/41bcd1b0-8a3b-46ba-a12b-799f3e9b2f6b) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/140174 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/7594 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/122987 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/85791 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/9fcea437-6aa2-4438-a444-6dcc182fa317) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/7231 "Passed tests") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/4944 "Passed tests") | [✅ 🛠 wpe-cairo-libwebrtc](https://ews-build.webkit.org/#/builders/166/builds/5567 "Built successfully") | | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/129190 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/116590 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/41145 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/147736 "Built successfully") | | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/135726 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/9272 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/41707 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/113309 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/9290 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/7805 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/113640 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/28862 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/7151 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/119237 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/63762 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/9321 "Built successfully") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/37291 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/168497 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/9047 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/72886 "Built successfully") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/43984 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/9261 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/9113 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

kmiller68 commented on Source/JavaScriptCore/yarr/generateYarrUnicodePropertyTables.py:
```diff
@@ -565,7 +565,7 @@ def convertStringToCppFormat(self, file, utf32String):
             else:
                 output += "\\x{:x}".format(ord(codePoint))
-        file.write("{{ std::span {{ const_cast<char32_t*>(U\"{}\"), {} }}}}".format(output, len(utf32String)))
+        file.write("{{ std::span {{ U\"{}\", {} }}}}".format(output, len(utf32String)))
```
That's a lot of curly braces lol

cdumez replied:
The double braces are just Python escaping. The `{}` are for parameters. The output looks fine to me?
```
{ std::span { U"\x1f9d1\x200d\x1f9b2", 3 }},
```

kmiller68 commented on Source/WTF/wtf/Vector.h:
```diff
@@ -163,24 +160,17 @@ struct VectorTypeOperations
     }
 
-    static void uninitializedCopy(const T* src, const T* srcEnd, T* dst)
+    template<std::size_t Extent = std::dynamic_extent>
```
Nit: Do you need the default value? Isn't it going to be deduced by the span passed as the first argument anyway?

cdumez replied:
Probably not needed, I'll try and remove it.

kmiller68 reviewed (approved):
r=me with nit.

cdumez committed:
Update Vector's uninitializedCopy() to use std::span
https://bugs.webkit.org/show_bug.cgi?id=304756

Reviewed by Keith Miller.

* Source/JavaScriptCore/yarr/generateYarrUnicodePropertyTables.py:
(PropertyData.convertStringToCppFormat):
* Source/WTF/wtf/Vector.h:
(WTF::VectorCopier::uninitializedCopy):
(WTF::VectorTypeOperations::uninitializedCopy):
(WTF::Vector::Vector):
(WTF::Malloc>::Vector):
(WTF::=):
(WTF::Malloc>::append):
(WTF::Malloc>::unsafeAppendWithoutCapacityCheck):
(WTF::Malloc>::insertSpan):

Canonical link: https://commits.webkit.org/305031@main
